bin/roave-infection-static-analysis-plugin

```diff
@@ -56,10 +57,61 @@ use function var_export;
         define('PHP_PARSER_VERSION', Versions::getVersion('nikic/php-parser'));
     }
 
-    $makeAnalyzer = static function () use ($projectPath): ProjectAnalyzer {
+    $extractArgument =
+        /**
+         * @param list<string> $arguments
+         *
+         * @return array{0: list<string>, 1: string|null}
+         */
+        static function(array $arguments, string $arg): array {
+            $lookup = '--'.$arg;
+
+            $result = null;
+            foreach ($arguments as $index => $argument) {
+                if ($argument === $lookup) {
+                    unset($arguments[$index]);
+                    $value = $arguments[$index+1] ?? null;
+                    if (null !== $value && !str_starts_with($value, '-')) {
+                        $result = $value;
+                        unset($arguments[$index+1]);
+                    }
+
+                    break;
+                }
+
+                if (str_starts_with($argument, '--psalm-config=')) {
+                    unset($arguments[$index]);
+
+                    $result = substr($argument, 15);
+                    break;
+                }
+            }
+
+            $arguments = array_values($arguments);
+            if (null !== $result) {
+                if (str_starts_with($result, '"')) {
+                    $result = substr($result, 1);
+                }
+
+                if (str_ends_with($result, '"')) {
+                    $result = substr($result, -1);
+                }
+            }
+
+            return [$arguments, $result];
+        };
+
+    [$argv, $configuration] = $extractArgument($_SERVER['argv'], 'psalm-config');
```

Comment:
If I understand this correctly, this is only about adding `--psalm-config`? If so, this patch seems simple enough 👍

Reply:
yea, this one is just adding `psalm-config` flag, given `(['--foo', '--psalm-config' 'bar'], 'psalm-config')` it returns `(['--foo'], 'bar')`.